packages/block-library/src/image/index.php

```diff
@@ -27,16 +28,88 @@ function render_block_core_image( $attributes, $content ) {
 		// which now wraps Image Blocks within innerBlocks.
 		// The data-id attribute is added in a core/gallery `render_block_data` hook.
 		$processor->set_attribute( 'data-id', $attributes['data-id'] );
+	}
+
+	$link_destination = isset( $attributes['linkDestination'] ) ? $attributes['linkDestination'] : 'none';
+
+	// Get the lightbox setting from the block attributes.
+	if ( isset( $attributes['behaviors']['lightbox'] ) ) {
+		$lightbox = $attributes['behaviors']['lightbox'];
+		// If the lightbox setting is not set in the block attributes, get it from the theme.json file.
+	} else {
+		$theme_data = WP_Theme_JSON_Resolver_Gutenberg::get_merged_data()->get_data();
+		if ( isset( $theme_data['behaviors']['blocks']['core/image']['lightbox'] ) ) {
+			$lightbox = $theme_data['behaviors']['blocks']['core/image']['lightbox'];
+		} else {
+			$lightbox = false;
+		}
+	}
+
+	$experiments = get_option( 'gutenberg-experiments' );
+
+	if ( ! empty( $experiments['gutenberg-interactivity-api-core-blocks'] ) && 'none' === $link_destination && $lightbox ) {
+
+		$aria_label = 'Open image lightbox';
+		if ( $processor->get_attribute( 'alt' ) ) {
+			$aria_label .= ' - ' . $processor->get_attribute( 'alt' );
+		}
 		$content = $processor->get_updated_html();
+
+		// Wrap the image in the body content with a button.
+		$img = null;
+		preg_match( '/<img[^>]+>/', $content, $img );
+		$button       = '<div class="img-container">
+			 					<button aria-haspopup="dialog" aria-label="' . $aria_label . '" data-wp-on.click="actions.core.image.showLightbox"></button>'
+									. $img[0] .
+								'</div>';
+		$body_content = preg_replace( '/<img[^>]+>/', $button, $content );
+
+		// For the modal, set an ID on the image to be used for an aria-labelledby attribute.
+		$modal_content = new WP_HTML_Tag_Processor( $content );
+		$modal_content->next_tag( 'img' );
+		$image_lightbox_id = $modal_content->get_attribute( 'class' ) . '-lightbox';
+		$modal_content->set_attribute( 'id', $image_lightbox_id );
+		$modal_content = $modal_content->get_updated_html();
+
+		$background_color  = wp_get_global_styles( array( 'color', 'background' ) );
+		$close_button_icon = '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="30" height="30" aria-hidden="true" focusable="false"><path d="M13 11.8l6.1-6.3-1-1-6.1 6.2-6.1-6.2-1 1 6.1 6.3-6.5 6.7 1 1 6.5-6.6 6.5 6.6 1-1z"></path></svg>';
+
+		return
+			<<<HTML
+				<div class="wp-lightbox-container"
+					data-wp-island=""
```

Comment:
Not a big deal, but I believe we could use `data-wp-island` instead of `data-wp-island=""` because it is a [boolean attribute](https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML).